### PR TITLE
fix hodir and leviathan

### LINKED ISSUE #6033: (Scripts/Ulduar): Flame Leviathan hard mode boss fight issue
**CURRENT BEHAVIOUR:**

Unable to start encounter Flame Leviathan hard mode boss fight, after talking with Lore Keeper of Norgannon.

**EXPECTED BLIZZLIKE BEHAVIOUR:**

After talking with Lore Keeper of Norgannon, it should encounter Flame Leviathan hard mode boss fight.

**SOURCE:**

**STEPS TO REPRODUCE THE PROBLEM:**

1. .go c 136605
2. talk to Lore Keeper of Norgannon
3. do the Flame Leviathan boss fight

**EXTRA NOTES:**

**AC HASH/COMMIT:**
 
a9e3d8d13ac6177772aff130ceef3570174bc3df

**OPERATING SYSTEM:**

win10

**MODULES:**

mod-ah-bot
mod-anticheat
mod-cfbg
mod-eluna-lua-engine
mod-individual-xp
mod-transmog

**CUSTOMIZATIONS:**

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/98658096-scripts-ulduar-flame-leviathan-hard-mode-boss-fight-issue?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### LINKED ISSUE #8168: Hodir 10 man Hardmode loot
### Current Behaviour

When you kill Hodir in 10 man Hardmode he spawns the Rare Cache of Winter. However in the Chest is 25 man Hardmode loot, and not the 10 man loot.

### Expected Blizzlike Behaviour

When you kill him in 10 man Hardmode you get 10 man Hardmode loot and nor 25.

### Source

_No response_

### Steps to reproduce the problem

Enter ulduar 10 man
Kill Hodir in Hardmode
Loot Rare Cache of Winter

### Extra Notes

_No response_

### AC rev. hash/commit

 AzerothCore rev. b2f08934bcc6 2021-09-27 22:15:03 +0000 (master branch) (Uni
x, RelWithDebInfo, Static)
Using SSL version: OpenSSL 1.1.1f  31 Mar 2020 (library: OpenSSL 1.1.1f  31 Mar
2020)
Using Boost version: 1.71.0
Using MySQL version: 80026
Using CMake version: 3.16.3
Compiled on: Linux 5.4.72-microsoft-standard-WSL2
Worldserver listening connections on port 8085
Realmlist (Realm Id: 1) configured in port 8085
VMAPs status: Enabled. LineOfSight: 1, getHeight: 1, indoorCheck: 1
MMAPs status: Enabled
maps directory located in /azerothcore/env/dist/data/maps. Total size: 252191207
 bytes
vmaps directory located in /azerothcore/env/dist/data/vmaps. Total size: 6581307
21 bytes
mmaps directory located in /azerothcore/env/dist/data/mmaps. Total size: 2196365
032 bytes
Using enUS DBC Locale as default. All available DBC locales: enUS
Using World DB: ACDB 335.4-dev
Using World DB Revision: 2021_09_27_04
Using Character DB Revision: 2021_09_25_00
Using Auth DB Revision: 2021_06_17_00

### Operating system

Win 10 running with docker

### Custom changes or Modules

noone

### PR DESCRIPTION
**Pull request by bgcfb**
## Changes Proposed:
-  fix hodir 10/25 mode chest and achievement
-  fix hodir hard mode miss but player can get hard mode chest
-  fix hodir npc mage toasty fire time line
-  fix leviathan start status

## Issues Addressed:

- Closes #8168
- Closes #6033

## Tests Performed:

- hodir chest should be correct
- hodir chest achievement can get now
- Flame Leviathan hard mode shoud work

## How to Test the Changes:

hodir:
1.go c 136262
2.start combat and wait hard mode miss
3.kill the boss and loot

Flame Leviathan:
1.go c 136605
2.talk and start hard mode

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by bgcfb** [comment]
It's my first time to make pr,pls check

**Comment by Nyeriah** [comment]
Please include what tests have you performed and how users can test the changes you have made in the pull request description.

**Comment by bgcfb** [comment]
> Please include what tests have you performed and how users can test the changes you have made in the pull request description.

It's fixed a month ago and I'am running now with no error

File: data/sql/updates/pending_db_world/rev_1633001158089619377.sql.sql [hunk 1]
```diff
@@ -0,0 +1,9 @@
...
+(33686, 0, 9, 16, 61, 0, 100, 0, 10477, 0, 0, 0, 0, 83, 1, 0, 0, 0, 0, 0, 10, 136253, 33579, 0, 0, 0, 0, 0, 0, 'Lore Keeper of Norgannon - On Gossip Option 0 Selected - Remove Npc Flags Gossip');
+
+INSERT INTO smart_scripts (entryorguid, source_type, id, link, event_type, event_phase_mask, event_chance, event_flags, event_param1, event_param2, event_param3, event_param4, event_param5, action_type, action_param1, action_param2, action_param3, action_param4, action_param5, action_param6, target_type, target_param1, target_param2, target_param3, target_param4, target_x, target_y, target_z, target_o, comment) VALUES
+(33686, 0, 16, 0, 61, 0, 100, 0, 10477, 0, 0, 0, 0, 45, 1, 1, 0, 0, 0, 0, 10, 136253, 33579, 0, 0, 0, 0, 0, 0, 'Lore Keeper of Norgannon - On Gossip Option 0 Selected - Set Data 1 1');
```
Comment (FrancescoBorzi) [inline data/sql/updates/pending_db_world/rev_1633001158089619377.sql.sql:9]: Hello, thanks for your contribution! You can combine the 2 inserts in one. See https://www.azerothcore.org/wiki/sql-standards

**Comment by acidmanifesto** [comment]
@bgcfb what is the current status of this pr?

**Comment by bgcfb** [comment]
> @bgcfb what is the current status of this pr?

Sql check fail

File: data/sql/updates/pending_db_world/rev_1633001158089619377.sql [hunk 2]
```diff
@@ -0,0 +1,8 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1633001158089619377');
+
+DELETE FROM `smart_scripts` WHERE `entryorguid` = 33686 AND `source_type` = 0 AND `id` = 9;
+
+INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `event_param5`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_param4`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
+(33686, 0, 9, 16, 61, 0, 100, 0, 10477, 0, 0, 0, 0, 83, 1, 0, 0, 0, 0, 0, 10, 136253, 33579, 0, 0, 0, 0, 0, 0, 'Lore Keeper of Norgannon - On Gossip Option 0 Selected - Remove Npc Flags Gossip'),
+(33686, 0, 16, 0, 61, 0, 100, 0, 10477, 0, 0, 0, 0, 45, 1, 1, 0, 0, 0, 0, 10, 136253, 33579, 0, 0, 0, 0, 0, 0, 'Lore Keeper of Norgannon - On Gossip Option 0 Selected - Set Data 1 1');
+
```
Comment (claudiodfc) [inline data/sql/updates/pending_db_world/rev_1633001158089619377.sql:8]: remove this line for SQL checks

**Comment by claudiodfc** [comment]
https://pastebin.com/TV0SrhE5 take a look into the crash if you will

**Comment by bgcfb** [comment]
> https://pastebin.com/TV0SrhE5 take a look into the crash if you will

New PR break ulduar and it is not related to my fix
I run this fix for month with no error

File: src/server/scripts/Northrend/Ulduar/Ulduar/instance_ulduar.cpp [hunk 3]
```diff
@@ -691,8 +691,8 @@ class instance_ulduar : public InstanceMapScript
                     break;
 
                 case TYPE_SPAWN_HODIR_CACHE:
-                    // Is the difficulty 10 man(0) ? return 10 man : return 25 man;
-                    SpawnHodirChests(m_difficulty == 0 ? 0 : 1);
+                    // Is the difficulty 10 man(1) ? return 10 man : return 25 man;
+                    SpawnHodirChests(m_difficulty == 0 ? 1 : 0);
```
Comment (claudiodfc) [inline src/server/scripts/Northrend/Ulduar/Ulduar/instance_ulduar.cpp:695]: change to
``` c++
SpawnHodirChests(m_difficulty);
```

**Review by claudiodfc — CHANGES_REQUESTED** [review]

**Comment by acidmanifesto** [comment]
What is the status of this PR? Is it abandoned now?

**Pull request closed without merging**